client/connection/openvpn.go

```diff
@@ -57,3 +64,18 @@ func ConfigureVpnClientFactory(
 		), nil
 	}
 }
+
+func detectCountry(ipResolver ip.Resolver, locationDetector location.Detector) (string, error) {
+	ip, err := ipResolver.GetPublicIP()
+	if err != nil {
+		return "", errors.New("IP detection failed: " + err.Error())
```

Comment:
Simply return error, don't wrap it into new one.

Reply:
done